beacon-chain/sync/data_columns.go

```diff
@@ -174,7 +237,316 @@ func RequestDataColumnSidecarsByRoot(
 	}
 
 	// If we still have remaining columns after all retries, return error
-	return nil, errors.Errorf("failed to retrieve all requested data columns after retries for block root=%#x, missing columns=%v", blockRoot, uint64MapToSortedSlice(remainingMissingColumns))
+	return nil, fmt.Errorf("failed to retrieve all requested data columns after retries for block root=%#x, %w", blockRoot, NewUnavailableColumnsError(uint64MapToSortedSlice(remainingMissingColumns)))
```

Comment:
Why removing the `missing columns` in logs?

Reply:
It's a bit of resources wasting here:
If only one data column is missing, we throw all of them.
I think this function could return, for example 3 items:
1. The verified RO sidecars returned
2. A slice of indices of missing data columns after this function run (in the happy case, an empty slice)
3. An error. This error would still be nil if some data columns are missing at the end of the function.

Reply:
If, for example, you are already able to retrieve 7 columns and 1 is still missing, then, in order to reconstruct, you need only to retrieve 64-7 = 57 columns to get the initial column you want, instead of 64.

Reply:
ad50969755ebde47e3ba8c4d004fe98ab2810ff6 (ByRoot) and 0857410b1eaa4a251ebcc7fd2aa19b31a3cd60ce (ByRange) now return partial results from the internal helper functions to allow the columns to be reused during reconstruction. These partial results are not yet used, still working on this.

Reply:
[62572a8](https://github.com/OffchainLabs/prysm/pull/15107/commits/62572a8fbb815d9a4eae6f5ad16758db99f05c40)